docs/apps/components/sub-form.md

```diff
@@ -0,0 +1,148 @@
+# Sub-Form
+
+## Description
+
+A **Sub-Form**, is a powerful tool in form creation that allows you to embed one form within another.
+
+Think of it as a smaller form placed within the larger form, providing additional details or related information about a specific section.
+
+Sub-form component, for list (array) type form data, uses Table layout, doesn't support layout nesting.
+
+<img src= "/apps/components/img/subform.png">
+
+## Use Case
+
+1. Improved organization
+2. Reduced redundancy
+3. Flexibility and customization
+4. Enhanced data accuracy
+5. Order forms
+6. Employee information forms
+7. Project management forms
+8. Survey forms
+9. Customizable product configuration
+10. Team member contributions
+11. Feedback aggregation
+12. Hierarchical data structures
+
+### Interactivity
+
+1. Show or hide sub-forms based on user choices in the main form.
+2. Dynamically reveal additional sub-forms as users complete previous ones.
+3. Implement branching logic based on sub-form data.
+4. Automatically populate sub-form fields based on data entered in the main form.
+5. Enable data manipulation within sub-forms.
+6. Use sub-forms as dynamic filters or search features.
+7. Integrate drag-and-drop functionality for reordering or prioritizing sub-forms.
+8. Implement collaborative editing for shared sub-forms.
+9. Use progressive disclosure to reveal sub-form content selectively.
+
+## API
+
+### Events
+
+| **Name**| **Description**|
+|---------|----------------|
+|`onChange (element)`|The onChange event occurs when the value of a field is changed|
+|`onRowAdd (element)`|The onRowAdd event occurs when a row is added|
+|`onRowRemove (element)`|The onRowRemove event occurs when a row is removed|
+|`onPageChange (element)`|The onPageChange event occurs when a page is changed|
+
+### Methods
```

Comment:
Please add the setData method an exmplae is `this.setData({ subformId: [ { inputId: '1234', inputId2: 'value2' } ] })`
Also add getValue e.g `this.getValue('subformId')`
Also add getValues function which would be this.getValues()